src/raftstore/store/snap.rs

```diff
@@ -61,10 +115,82 @@ impl Display for SnapKey {
     }
 }
 
-/// Name prefix for the self-generated snapshot file.
-const SNAP_GEN_PREFIX: &'static str = "gen";
-/// Name prefix for the received snapshot file.
-const SNAP_REV_PREFIX: &'static str = "rev";
+pub trait SendSnapshotFileReader: Read {
+    fn exists(&self) -> bool;
+    fn path(&self) -> String;
+    fn meta(&self) -> io::Result<Metadata>;
+    fn total_size(&self) -> io::Result<u64>;
+    fn delete(&self);
+}
+
+pub trait RecvSnapshotFileReader {
+    fn path(&self) -> String;
+    fn exists(&self) -> bool;
+    fn delete(&self);
+    fn apply(&mut self,
+             db: Arc<DB>,
+             region: &Region,
+             abort: Arc<AtomicUsize>,
+             write_batch_size: usize,
+             snapshot_size: &mut usize,
+             snapshot_kv_count: &mut usize)
+             -> Result<()>;
+}
+
+pub trait SnapshotFileWriter: Write + Send {
+    fn path(&self) -> String;
+    fn exists(&self) -> bool;
+    fn save(&mut self) -> io::Result<()>;
+}
+
+pub struct SnapFileReader {
```

Comment:
It's doable, however not quite pithy to add V2, V3 for new impl later, even though maybe there will be only one Vn as usual.
I think it's better to separate different format impl in different module to avoid the name clashes, and use them like `v1::xxxReader`, `v2::xxxReader` in `snap.rs`, outside `snap.rs`, there would be only trait name but no impl class seen.